### ISSUE TITLE
v6.2?

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#1604]
How about planning for a 6.2 release? 

I am currently working on #188, which should help with #1582. Afterwards we could stress-test the new infrastructure with a new release to get rid of all the Python issues and to manifest the absence of MATLAB code. 

Any comments anyone? Who can help to gather the "must fix" issues for the next release?

**Comment by henningjp** [comment]
I'm all for it.  Let me know if I can help in any way.  I want to get the updates to the IF97 Backend and accompanying docs into an official release.  I have a couple of items I'm working on that should be no problem getting done before a 6.2 release.

**Comment by jowr** [comment]
OK, I'll draft a milestone github, please add relevant issue to it.

**Comment by thorade** [comment]
Should we also go through all the old issues and try to close some more?

**Comment by jowr** [comment]
Yes please, either close or assign to this milestone...

**Comment by henningjp** [comment]
Only two days left...  still a number of unresolved issues assigned.

**Comment by henningjp** [comment]
There are just nine open items on this milestone for release v6.2.0 (although some may be appreciable tasks).  I've got two of them that I believe I can complete by May 1, 2018.  Can we get an assignee on each of the remaining seven and an expected completion date (ECD) on each?  Alternatively, make the call that an issue won't be in this release.  This decision should be made on #1159 as well.  Thanks!

**Comment by ibell** [comment]
I think we finally have python 3.6 binary wheels for all systems.  That was a blocking issue and I see not major blocking issues now with a 6.2 release

**Comment by henningjp** [comment]
Issue #1678 removed from this release.  Should be ready to go.

**Comment by thorade** [comment]
Should we try to update some of the externals for the 6.2 release?
Some have not been update for 2 years.

**Comment by ibell** [comment]
You are welcome to experiment locally, but be careful - some API changes
have forced us to revert when we have bumped git submodules.  Before we
commit anything to master, let's run all the builds to make sure things are
ok.

**Comment by thorade** [comment]
Should we add Python 3.7 wheels for the 6.2 release?

**Comment by ibell** [comment]
Good idea! Does conda do python 3.7?  That's a requirement before we can add the wheels

**Comment by thorade** [comment]
Not sure. The downloadable installer still uses 3.6, 
https://www.anaconda.com/download/
but there is a python 3.7 package:
https://anaconda.org/anaconda/python

**Comment by jowr** [comment]
I am fine with skipping 6.1.1 and moving straight to 6.2 to include the new features mentioned in #1723. I have added all Python 3.7 builders and will merge the changes to the master later today. Talking about updates, I can see two more things that we might want to consider:

- [ ] Update buildbot system 
  - [x] Docker images are updated and thus incompatible with the current build master server
  - [ ] Update the server (@jowr)
  - [ ] Update buildbot on "OSX-slave" (@ibell)
  - [ ] Update buildbot on "linux-slave" (@ibell)
  - [ ] Update buildbot on "windows-slave" (@ibell)
- [ ] check if externals should be updated (could that be you @thorade ?)

**Comment by thorade** [comment]
Thanks for the py37 builders!
I can play with the externals, but I understood I need to run some test builds locally and would need advise on how to do that.

**Comment by jowr** [comment]
It should all be good to go using TravisCI https://travis-ci.org/CoolProp/CoolProp - make a branch, update externals and keep an eye on TravisCI. If things work there, let me know and I can fire up the buildbots using your branch. Our builders are quite slow and they do not process branches automatically...

**Comment by thorade** [comment]
OK, I'll add a branch and do the updates there.
I also did that once in the past, in PR #1622 and Travis accepted that.

**Comment by thorade** [comment]
The following submodules can NOT be updated to latest master:
- cppformat (used the commit that was successful in #1622 instead)
- catch (used v2.2.3 instead) 
-REFPROP-headers (not updated)

Everything else has been updated to latest master in the branch.
For the future, it might make sense to not checkout latests master, but latest release!?

**Comment by ibell** [comment]
Can we make small changes to make those submodules happy, or is it a lost
cause?

**Comment by thorade** [comment]
The builds and log are here:
https://travis-ci.org/CoolProp/CoolProp/builds

For REFPROP-headers, it should be possible to move to the latest master. The commit used is 20 commits behind the master, all 20 commits are from Ian. Could you look at this log?
https://travis-ci.org/CoolProp/CoolProp/jobs/416989963

Catch is using the second-newest release, the latest release mentions that include paths have changed:
https://github.com/catchorg/Catch2/releases
So for Catch I see good chances to just adapt the include paths.

For cppformat, the first error is that one file is missing/moved. But by now, it is 600+ commits behind, so there might be some other changes.

**Comment by jowr** [comment]
We are preparing for new release and we are not going to update the buildbot infrastructure this time. As soon as the Windows builders work properly, we are going to make a new release. Any features that should be included?

**Comment by henningjp** [comment]
Trying to get the implementation of CoolProp dipole moments worked out.  We have a few materials that need better references to get them in tighter agreement with REFPROP values.  I think this is a worthwhile addition. I’ll tag it. PR #1701
